### PR TITLE
sweep_nulls function tests

### PR DESCRIPTION
**Pull request by y1chi-z**
1. create `tests/test_sweep_nulls.py`: test cases for `sweep_nulls` function
2. update `src/datamop/sweep_nulls.py` : function implementation

### CODE REVIEW

File: tests/test_sweep_nulls.py [hunk 1]
```diff
@@ -0,0 +1,279 @@
+import pytest
+import pandas as pd
+import numpy as np
+import sys
+import os
+sys.path.append(os.path.join(os.path.dirname(__file__), '..'))
```
Comment (davyxuximin) [inline tests/test_sweep_nulls.py]: As we update the init file, you only need `import datamop` to get access to your function

File: tests/test_sweep_nulls.py [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+
+# Error Case 1: Unsupported data type
+def test_unsupported_data_type():
+    """
+    Tests unsupported data types.
+    A ValueError is raised for inputs that are not pandas DataFrames.
+    """
+    with pytest.raises(ValueError, match="Input data must be a pandas DataFrame"):
+        datamop.sweep_nulls("not a dataframe", strategy='mean')
+        datamop.sweep_nulls(pd.Series([1, 2, np.nan]), strategy='mean')
```
Comment (davyxuximin) [inline tests/test_sweep_nulls.py]: `with pytest.raises` will exit after the first exception, the second one will never reached, try to put them in seperate function

**Review by davyxuximin — CHANGES_REQUESTED** [review]
The tests overall looks good, but need improvement. a 98% coverage is achieved which is fairly good

File: tests/test_sweep_nulls.py [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+
+# Edge Cases 8: Non-numeric columns with numeric strategies
+def test_non_numeric_with_numeric_strategy(sample_data):
+    """
+    Tests applying a numeric strategy to non-numeric columns
+    A warning is raised and no changes are made to the column.
+    """
+    with pytest.warns(UserWarning, match="Strategy 'mean' cannot be applied to non-numeric column 'c'"):
+        datamop.sweep_nulls(sample_data, strategy='mean', columns=['c'])
+
```
Comment (sep-he) [inline tests/test_sweep_nulls.py:235]: Consider adding an assert to check that there are no changes made to the column after the warning.

File: src/datamop/sweep_nulls.py [hunk 4]
```diff
@@ -40,4 +57,74 @@ def sweep_nulls(data, strategy='mean', columns=None, fill_value=None):
...
+                    raise TypeError("Invalid `fill_value` type.")
+                data[column] = data[column].fillna(fill_value)
+            elif strategy == 'drop':
+                data = data.dropna(subset=[column])
+
+    # Restore the original data types
+    remaining_columns = data.columns
+    for column in remaining_columns:
+        if column in original_dtypes:
+            data[column] = data[column].astype(original_dtypes[column])
```
Comment (sep-he) [inline src/datamop/sweep_nulls.py]: Consider adding a try except block here. 
`try: data[column] = data[column].astype(original_dtypes[column])`
`except ValueError: warnings.warn("message about not restoring the type", UserWarning)`

**Review by sep-he — COMMENTED** [review]
Looks good.

Perhaps handle the case when DataFrame is empty, there would be a warning message, and update Edge Case 1 test.

**Review by sep-he — APPROVED** [review]
Looks good.

**Review by davyxuximin — APPROVED** [review]
Problem fixed
